// ngx-highlight-js/bundles/highlightjs.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/forms')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@angular/common', '@angular/forms'], factory) :
	(factory((global.ngxHighlightJS = {}),global.ng.core,global.ng.common,global.ng.forms));
}(this, (function (exports,core,common,forms) { 'use strict';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var HighlightJsDirective = /** @class */ (function () {
    function HighlightJsDirective(el, doc) {
        this.el = el;
        this.doc = doc;
    }
    /**
     * @param {?} str
     * @return {?}
     */
    HighlightJsDirective.prototype.escapeHTML = /**
     * @param {?} str
     * @return {?}
     */
    function (str) {
        return (str || '')
            .replace(/&/g, '&amp;')
            .replace(/</g, '&lt;')
            .replace(/>/g, '&gt;')
            .replace(/"/g, '&quot;')
            .replace(/'/g, '&apos;');
    };
    /**
     * @return {?}
     */
    HighlightJsDirective.prototype.init = /**
     * @return {?}
     */
    function () {
        this.destroy();
        this.codeEl = this.doc.createElement('pre');
        if (this.lang) {
            this.codeEl.className = this.lang;
        }
        this.codeEl.innerHTML =
            this.code || '' + this.el.nativeElement.innerHTML.trim();
        this.parentEl = this.el.nativeElement.parentNode;
        this.parentEl.insertBefore(this.codeEl, this.el.nativeElement.nextSibling);
        hljs.configure(Object.assign({}, this.options));
        hljs.highlightBlock(this.codeEl);
    };
    /**
     * @return {?}
     */
    HighlightJsDirective.prototype.destroy = /**
     * @return {?}
     */
    function () {
        if (this.codeEl) {
            this.parentEl.removeChild(this.codeEl);
            this.codeEl = null;
        }
    };
    /**
     * @return {?}
     */
    HighlightJsDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.init();
    };
    /**
     * @return {?}
     */
    HighlightJsDirective.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this.ngModel) {
            this.ngModel.valueChanges.subscribe(function (res) {
                _this.code = _this.escapeHTML(res);
                _this.init();
            });
        }
    };
    /**
     * @return {?}
     */
    HighlightJsDirective.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.destroy();
    };
    HighlightJsDirective.decorators = [
        { type: core.Directive, args: [{
                    selector: '[highlight-js]',
                    host: {
                        style: 'display:none;',
                    },
                },] },
    ];
    /** @nocollapse */
    HighlightJsDirective.ctorParameters = function () { return [
        { type: core.ElementRef },
        { type: undefined, decorators: [{ type: core.Inject, args: [common.DOCUMENT,] }] }
    ]; };
    HighlightJsDirective.propDecorators = {
        options: [{ type: core.Input }],
        lang: [{ type: core.Input }],
        code: [{ type: core.Input }],
        ngModel: [{ type: core.ContentChild, args: [forms.NgModel,] }]
    };
    return HighlightJsDirective;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var HighlightJsModule = /** @class */ (function () {
    function HighlightJsModule() {
    }
    HighlightJsModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [forms.FormsModule],
                    declarations: [HighlightJsDirective],
                    exports: [HighlightJsDirective]
                },] },
    ];
    return HighlightJsModule;
}());

exports.HighlightJsDirective = HighlightJsDirective;
exports.HighlightJsModule = HighlightJsModule;

Object.defineProperty(exports, '__esModule', { value: true });

})));
