// ngx-highlight-js/esm5/highlightjs.js
import { ContentChild, Directive, ElementRef, Inject, Input, NgModule } from '@angular/core';
import { DOCUMENT } from '@angular/common';
import { FormsModule, NgModel } from '@angular/forms';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var HighlightJsDirective = /** @class */ (function () {
    function HighlightJsDirective(el, doc) {
        this.el = el;
        this.doc = doc;
    }
    /**
     * @param {?} str
     * @return {?}
     */
    HighlightJsDirective.prototype.escapeHTML = /**
     * @param {?} str
     * @return {?}
     */
    function (str) {
        return (str || '')
            .replace(/&/g, '&amp;')
            .replace(/</g, '&lt;')
            .replace(/>/g, '&gt;')
            .replace(/"/g, '&quot;')
            .replace(/'/g, '&apos;');
    };
    /**
     * @return {?}
     */
    HighlightJsDirective.prototype.init = /**
     * @return {?}
     */
    function () {
        this.destroy();
        this.codeEl = this.doc.createElement('pre');
        if (this.lang) {
            this.codeEl.className = this.lang;
        }
        this.codeEl.innerHTML =
            this.code || '' + this.el.nativeElement.innerHTML.trim();
        this.parentEl = this.el.nativeElement.parentNode;
        this.parentEl.insertBefore(this.codeEl, this.el.nativeElement.nextSibling);
        hljs.configure(Object.assign({}, this.options));
        hljs.highlightBlock(this.codeEl);
    };
    /**
     * @return {?}
     */
    HighlightJsDirective.prototype.destroy = /**
     * @return {?}
     */
    function () {
        if (this.codeEl) {
            this.parentEl.removeChild(this.codeEl);
            this.codeEl = null;
        }
    };
    /**
     * @return {?}
     */
    HighlightJsDirective.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.init();
    };
    /**
     * @return {?}
     */
    HighlightJsDirective.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (this.ngModel) {
            this.ngModel.valueChanges.subscribe(function (res) {
                _this.code = _this.escapeHTML(res);
                _this.init();
            });
        }
    };
    /**
     * @return {?}
     */
    HighlightJsDirective.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.destroy();
    };
    HighlightJsDirective.decorators = [
        { type: Directive, args: [{
                    selector: '[highlight-js]',
                    host: {
                        style: 'display:none;',
                    },
                },] },
    ];
    /** @nocollapse */
    HighlightJsDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
    ]; };
    HighlightJsDirective.propDecorators = {
        options: [{ type: Input }],
        lang: [{ type: Input }],
        code: [{ type: Input }],
        ngModel: [{ type: ContentChild, args: [NgModel,] }]
    };
    return HighlightJsDirective;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var HighlightJsModule = /** @class */ (function () {
    function HighlightJsModule() {
    }
    HighlightJsModule.decorators = [
        { type: NgModule, args: [{
                    imports: [FormsModule],
                    declarations: [HighlightJsDirective],
                    exports: [HighlightJsDirective]
                },] },
    ];
    return HighlightJsModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */

export { HighlightJsDirective, HighlightJsModule };
